### ISSUE TITLE
Unsupported model architecture for VanillaGradients

### ISSUE THREAD

**Issue by khaledbouabdallah** [sicara/tf-explain#182]
I'm trying to use VanillaGradients for a single image, but I'm getting this warning 

> UserWarning: Unsupported model architecture for VanillaGradients. The last two layers of the model should be: a layer which computes class scores with no activation, followed by an activation layer.

**VanillaGradients call :**

```python
img = tf.keras.preprocessing.image.load_img(image_path,target_size=IMG_SHAPE)
img = tf.keras.preprocessing.image.img_to_array(img)
img = np.array(img) / 255

data = ([img], None)

explainer = VanillaGradients()
grid = explainer.explain(data,model,0)
```

**Model Declaration**

```python
model = Sequential()
model.add(Conv2D(32 , (3,3) , strides = 1 , padding = 'same' , activation = 'relu' , input_shape = (150,150,3)))
model.add(BatchNormalization())
model.add(MaxPool2D((2,2) , strides = 2 , padding = 'same'))
model.add(Conv2D(64 , (3,3) , strides = 1 , padding = 'same' , activation = 'relu'))
model.add(Dropout(0.1))
model.add(BatchNormalization())
model.add(MaxPool2D((2,2) , strides = 2 , padding = 'same'))
model.add(Conv2D(64 , (3,3) , strides = 1 , padding = 'same' , activation = 'relu'))
model.add(BatchNormalization())
model.add(MaxPool2D((2,2) , strides = 2 , padding = 'same'))
model.add(Conv2D(128 , (3,3) , strides = 1 , padding = 'same' , activation = 'relu'))
model.add(Dropout(0.2))
model.add(BatchNormalization())
model.add(MaxPool2D((2,2) , strides = 2 , padding = 'same'))
model.add(Conv2D(256 , (3,3) , strides = 1 , padding = 'same' , activation = 'relu'))
model.add(Dropout(0.2))
model.add(BatchNormalization())
model.add(MaxPool2D((2,2) , strides = 2 , padding = 'same'))
model.add(Flatten())
model.add(Dense(units = 128 , activation = 'relu'))
model.add(Dropout(0.2))
model.add(Dense(units = 1 , activation = 'sigmoid'))
```

**Notes**

-  GRAND CAM and occlusion are working as expected but all gradient-based methods results in the same warning
- I'm working on binary classification

**Comment by E0HYL** [comment]
I met the same problem when running with a model where the last layer is `Dense` with an activation function.
I checked the paper of [Vinilla Gradients](https://arxiv.org/pdf/1312.6034.pdf). In Section 2, the authors pointed that
> It should be noted that we used the (unnormalised) class scores S_c, rather than the class posteriors, returned by the soft-max layer... The reason is that the maximisation of the class posterior can be achieved by minimising the scores of other classes.
Guess this is the reason why the warning is raised.
